### PR TITLE
refactor(webinstall): Reprompt after invalid input instead of exiting

### LINKED ISSUE #1561: Webinstall script: Prompt user to try again on invalid input
### Feature request summary

When installing kraftkit using the Webscript, if the user enters an invalid input the script throws an exception and exits. I think it would be nicer to just prompt the user to try again. For example, when I ran the installation script I made a typo and entered "zhs" instead of "zsh", which caused the script to exit immediately. If nobody has an objection to this I would like to contribute this change. 

### Describe alternatives

_No response_

### Related architectures

None

### Related platforms

None

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by HendrikHuebner**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [ ] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes
Closes  #1561

When a user enters something invalid in response to a prompt by the webinstall script, the script now displays a message such as: 

`Invalid input. Please try again or exit with Ctrl+C.`

Additionally, for yes/no questions I have created the prompt_user_yes_no function, which simplifies the script a little bit since it gets rid of some duplicated code.

### CODE REVIEW

File: tools/webinstall/install.sh [hunk 1]
```diff
@@ -199,6 +199,27 @@ get_user_response() {
...
+prompt_user_yes_no() {
+  while true;
+  do
+      get_user_response "$1" "$2"
+      _answer="$_RETVAL"
+
+      if printf "%s" "$_answer" | "$GREP" -q -E "$_NO_ANS"; then
+          return 1
+      elif printf "%s" "$_answer" | "$GREP" -q -E "$_YES_ANS"; then
+          return 0
```
Comment (craciunoiuc) [inline tools/webinstall/install.sh:216]: I think this does not catch default cases?

see `_NO_ANS_DEFAULT`/`_YES_ANS_DEFAULT`

**Review by craciunoiuc — CHANGES_REQUESTED** [review]
Hey @HendrikHuebner 

Looks really good, left you a comment. I still need to try it out.

Can you also reword your commit to start with `refactor(webinstall):` :pray: 

`refactor(webinstall): Prompt user again on invalid input instead of exiting`

Make sure you leave a single commit afterwards

File: tools/webinstall/install.sh (hunk 1 above)
Comment (HendrikHuebner) [reply tools/webinstall/install.sh:216]: I don't think those are actually needed. The default case (an empty string) is already handled with the second argument of get_user_response. The only thing that makes the `_NO_ANS_DEFAULT/_YES_ANS_DEFAULT `regexes different is that they match an empty string as well, so if I'm not mistaken that check was duplicated in the original script as well.

Comment (craciunoiuc) [reply tools/webinstall/install.sh:216]: They match also empty strings and only newlines

I'll try it like this now and see if it works

Comment (craciunoiuc) [reply tools/webinstall/install.sh:216]: Tried it out now

Seems to work, nice

**Review by craciunoiuc — CHANGES_REQUESTED** [review]
Seems that the commit is still to long 😅 

Try renaming it to, this one, it's 70 chars:

`refactor(webinstall): Reprompt after invalid input instead of exiting`

We can merge after

Previous one was `76` > `74` chars whoops

**Review by craciunoiuc — APPROVED** [review]
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
Approved-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

**Pull request merged**